core/drivers/imx_i2c.c

```diff
@@ -59,17 +62,24 @@ static struct io_pa_va i2c_bus[3] = {
 #if defined(I2C3_BASE)
 	[2] = { .pa = I2C3_BASE, },
 #endif
+#if defined(I2C4_BASE)
+	[3] = { .pa = I2C4_BASE, },
+#endif
 #endif
 };
 
 static struct imx_i2c_clk {
 	struct io_pa_va base;
 	uint32_t i2c[ARRAY_SIZE(i2c_bus)];
+#if defined(CFG_MX6ULL)
```

Comment:
This change doesn't seem related to the main purpose (add I2C4 support)? Therefore,  it should be moved to a separate commit.